sched/definitions.go

```diff
@@ -17,3 +17,33 @@ type Task struct {
 	Command    []string // Argv to run
 	SnapshotId string
 }
+
+type JobState int
+
+const (
+	// An unambiguous 0-value.
+	UNKNOWN JobState = iota
+	// Waiting to run.
+	PENDING
+	// Running
+	RUNNING
+	// Done
+	DONE
+)
+
+type JobStatus struct {
```

Comment:
I just meant make the struct type lower case.  that way they can't create on their own without using the factory method.  If you want to enforce immutability in the API you can make the struct components protected (lowercase) and expose accessor read only methods.